### PR TITLE
feat: initial port of v2 case card code

### PR DESCRIPTION
**Pull request by kswenson**
[[PT-187572179]](https://www.pivotaltracker.com/story/show/187572179)

This is a large PR -- here's some guidance for review:

1. Start with the documentation -- `case-card-port.md`
2. The bulk of the code is in the case card components themselves in `src/components/case-card/*.v2.js`. Assume that this code was previously reviewed in the v2 context and so should not be considered new code requiring full review. It's worth skimming these files to see how they're structured but don't get bogged down in the details.
3. It's probably worth spending more time with the jest tests than usual because they serve as a form of documentation for each tested component.
4. Similarly, some of the ported utilities (notably `dg-data-context-utilities.v2.js`) are not worth getting bogged down in the details, particularly in that case since much of the code is currently unused.
5. The compatibility code is where the bulk of the new development has occurred and is where detailed review is most warranted.

There will no doubt be a number of bugs, which should be added as additional PT stories.

### CODE REVIEW

**Review by bfinzer — APPROVED** [review]
👍🏻LGTM
Amazing work! I looked through everything. The compatibility code is very clear. I confess I don't understand how the tests work.

**Review by tealefristoe — APPROVED** [review]
Looks good! Quite the undertaking here. I hope importing the old code works as well as it seems it might. I left a smattering of suggestions.

**Comment by tealefristoe** [comment]
I guess one more thing to note is that the project code coverage drops by 14% with this PR. I don't really know what to make of the code coverage these days, so maybe it's not important.

File: v3/src/components/case-card/case-card.tsx [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+
+import { DG } from "../../v2/dg-compat.v2"
+import { DGDataContext } from "../../models/v2/dg-data-context"
+
+import "./case-card.v2"
+const DGCaseCard = (DG.React as any).CaseCard
 
 import "./case-card.scss"
+import { useMergeRefs } from "@floating-ui/react"
+import { useResizeDetector } from "react-resize-detector"
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/case-card/case-card.tsx:15]: I had already fixed this locally but forgot to push it up. 🤦

**Comment by kswenson** [comment]
>I guess one more thing to note is that the project code coverage drops by 14% with this PR. I don't really know what to make of the code coverage these days, so maybe it's not important.

I'm choosing not to worry about it to the extent that it holds up this PR further, but I don't want to ignore it completely, either. I can think of a few reasons why the code coverage would go down and what we might do about it, but I don't know that these ideas can explain a project-wide 14% drop, so there may be something else going on:
- there's a lot of drag-related code in the case card that isn't being used/tested
- `dg-data-context-utilities.v2.js` contains a lot of unused code that can/will eventually just be removed
- a more extensive cypress test could be written that tested some of the code not covered by the jest tests

In any case, it's something to consider moving forward but I don't think it's worth holding up this PR any longer.

**Pull request merged**